### DISCUSSION TITLE
How does dnsmasq configure network interfaces?

### DISCUSSION THREAD

**Discussion by RogueWarrior65** [RaspAP/raspap-webgui#875] [Q&A]
I realize that this question is likely out of the scope but I'll ask it because I've used an older, maybe two to three years, version of raspap and the current design is notably different.  The old version used to be able to configure network interfaces, eth0 en1sp0 wlan0, for either static or DHCP addressing as though you were doing an ifconfig command and modifying systemd network configuration files.  Now, changing network settings modifies dnsmasq.conf but how do those settings get used?  Does dnsmasq itself affect changes to the network interfaces?

**Comment by billz** [comment]
See the v2.6-beta [release notes](https://github.com/RaspAP/raspap-webgui/releases/tag/2.6-beta) for an explanation.

**Reply by RogueWarrior65** [reply]
Okay, so right now, my Yocto-based distro is using a systemd networking service to manage the IP address settings (rather than /etc/network/interfaces).  In the RaspAP world, should I disable this service?  If I do, what "component" then handles configuring network interfaces?  dnsmasq, dhcpd, hostapd?  Or something else?

**Reply by billz** [reply]
dhcpcd is used for network management. This is not the 'RaspAP world' but more accurately the RPi OS world, as systemd-networkd is disabled by default. The latter is only used by this project for network device creation, eg., bridged APs. See `installers/servicestart.sh`.

**Reply by RogueWarrior65** [reply]
Okay, that explains a lot.  dhcpcd wasn't installed.  I've gone ahead and installed it and it appears to be working for two ethernet interfaces however, it doesn't get an IP address for wlan0 when it's in DHCP mode.  Looking at the service status, I see "wlan0: waiting for 3rd party to configure IP address" which has me wondering who this mythical 3rd party is and why isn't dhcpcd handling this.

**Reply by RogueWarrior65** [reply]
I think I may have figured it out.  It appears that dhcpcd overrides any settings found in systemd-networkd service files.  This service wasn't running.  It's a little confusing that dhcpcd doesn't handle everything on its own even though it appears to do static addresses just fine.

**Comment by RogueWarrior65** [comment]
Okay, I'm 99 44/100ths% sure I can confirm that systemd-networkd needed to be enabled and running.  As a matter of interest, dhcpcd sets static IP addresses just fine but expects a "3rd party" to handle ones for DHCP.  Also, if the network adapter is unplugged, dhcpcd won't assign even a static IP address.  As soon as you plug it in though, it does.
